filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobPathBuilder.scala

```diff
@@ -90,4 +90,19 @@ case class BlobPath private[blob](nioPath: NioPath, endpoint: String, container:
   override def pathAsString: String = List(endpoint, container, nioPath.toString()).mkString("/")
 
   override def pathWithoutScheme: String = parseURI(endpoint).getHost + "/" + container + "/" + nioPath.toString()
+
+  def blobFileAttributes: Try[AzureBlobFileAttributes] =
+    Try(Files.readAttributes(nioPath, classOf[AzureBlobFileAttributes]))
+
+  def md5HexString: Try[Option[String]] = {
+    blobFileAttributes.map(h =>
+      Option(h.blobHttpHeaders().getContentMd5) match {
+        case None => None
+        case Some(arr) if arr.isEmpty => None
```

Comment:
This is really interesting, I hadn't realized you could pattern match like this!